vesc_hw_interface/src/vesc_servo_controller.cpp

```diff
@@ -301,6 +335,11 @@ void VescServoController::updateSensor(const std::shared_ptr<VescPacket const>&
     }
 
     sens_position_ -= getZeroPosition();
+
+    std::ofstream file;
+    file.open(calibration_result_path_, std::ios::out);
+    file << "servo/last_position: " << sens_position_ << std::endl;
+    file.close();
```

Comment:
ファイル書き出しは下記の最後ら辺に記述して、キャリブレーション終了時に一回の処理のほうがいいように思います
https://github.com/sbgisen/vesc/blob/dc4a7d44802af474ba26d996543488d413a2dce8/vesc_hw_interface/src/vesc_servo_controller.cpp#L256-L262

既存のキャリブレーション値を使用する場合、(コントローラがすでにそういう挙動になってるかもしれないですが)立ち上げ時に現在位置を初期の目標値にするため下記のような処理入れたほうが安全かなと思います
```cpp
target_position_ = target_position_previous_ = sens_position_;
```

Reply:
> ファイル書き出しは下記の最後ら辺に記述して、キャリブレーション終了時に一回の処理のほうがいいように思います

I want to save only calibration position too.
But, I don't think it would be possible without knowing where servo was finally located.
Because the control is based on the relative position of how much it has moved since it started.

Do you have any good idea?

Reply:
I understand what you want to do.

The value of _sens_position_ changes when the vesc is turned on again. if you save the zero_position_, it will need to be calibrated again when the vesc is rebooted.

If you want the ability to reboot vesc and not run calibration, you can save the last position you were in (unless you move it manually) and it will behave the way you want.

Reply:
We can't output ROS logger and `cout`, write to file in the destructor, when the node terminated with ctrl-c.
To be drafted once for investigation.

Reply:
I see. It would be great if you could disable file export when the path is empty.

Reply:
Sure. I'll fix it in [4088723](https://github.com/sbgisen/vesc/pull/81/commits/4088723661356c19f8a1e0e854eee7ed6c5a6b9d).

Reply:
>> please check the comment [#81 (comment)](https://github.com/sbgisen/vesc/pull/81#discussion_r1156848764)
>
> ファイル書き出しは下記の最後ら辺に記述して、キャリブレーション終了時に一回の処理のほうがいいように思います

I agree with this one.

> 既存のキャリブレーション値を使用する場合、(コントローラがすでにそういう挙動になってるかもしれないですが)立ち上げ時に現在位置を初期の目標値にするため下記のような処理入れたほうが安全かなと思います
> ```c++
> target_position_ = target_position_previous_ = sens_position_;
> ```

Regardless of what command (`target_position_` variable)  you initialize, ros_control will override so you will need an additional process or flag to remember the previous command. This is outside the scope of this PR so we will need to open issue to deal with this.

Reply:
Aren't you just giving that command?

Reply:
Are you saying that it will be overwritten **before the initialization phase**?

Reply:
Sorry, I overlooked the fact that `target_position_ = target_position_previous_ = sens_position_;` was not implemented here so forget about the comment above.